openapi/clusters_mgmt/v1/openapi.json

```diff
@@ -16427,23 +16431,87 @@
             }
           },
           "secret": {
-            "description": "The secret of the OIDC client from the OIDC provider.",
+            "description": "The secret of the OIDC client from the OIDC provider.\nThe client is considered 'public' if no secret is specified. Otherwise, it is considered\nas a 'confidential' client.\nThis can only be used for an external authentication provider belonging to a ROSA HCP cluster.",
             "type": "string"
+          },
+          "type": {
+            "description": "Determines the OIDC provider client type.\nThis is required to be defined for an external authentication provider belonging to an ARO HCP cluster.\n\nFor clients belonging to a ROSA HCP cluster, the value of this property will be determined by the \n'secret' property in the client configuration.",
```

Comment:
Here the challenge is that the `secret` attribute only exists for ROSA-HCP based clusters. ARO-HCP does not provide the secret through the `secret` API attribute because of security concerns.

For ROSA-HCP, will it be possible to explicitly define this attribute? or will it be a readonly one? is it possible to define the secret and it being "public"?